packages/option-t/src/plain_result/map_err_unknown.ts

```diff
@@ -0,0 +1,32 @@
+import { CausalCarrierError } from './internal/causal_carrier.js';
+import { mapErrForResult } from './map_err.js';
+import type { Result } from './result.js';
+
+export function mapErrUnknownToError<T>(input: Result<T, unknown>): Result<T, Error> {
+    const result = mapErrForResult<T, unknown, Error>(input, transformUnknownCausalToError);
+    return result;
+}
+
+function transformUnknownCausalToError(cause: unknown): Error {
+    if (cause instanceof Error) {
+        return cause;
+    }
+
+    const wrapper = new CausalCarrierError(cause);
+    return wrapper;
+}
+
+export function mapErrUnknownToCausalCarrierError<T>(
```

Comment:
Question: Should we need this?

Pros of this operator: Unify to wrap `E` by  `CausalCarrierError` always.
Cons of this operator: Create a redundant error chain.

Reply:
> Cons of this operator: Create a redundant error chain.

This looks like not attractive. But this might be problematic if we call this operator immediately after `tryCatchIntoResult()`. In such case, it's obviously the error causal place is "almost" same place. For other cases, shortcutting an error chain would be divide the error stack trace. It does not match with the manner that we said in https://github.com/option-t/option-t/pull/2280